koku/sources/test/test_kafka_listener.py

```diff
@@ -741,26 +738,3 @@ def test_sources_network_auth_info_unknown_provider(self):
         source_integration.sources_network_auth_info(test_resource_id, test_auth_header)
         source_obj = Sources.objects.get(source_id=test_source_id)
         self.assertEquals(source_obj.authentication, {})
-
-    async def test_listen_for_messages_error(self):
```

Comment:
since we are still collecting the kafka connection error metrics - should we modify this test to conform to the modified `listen_for_messages` function instead of just removing it?

Reply:
I did try to modify it, but that counter is now in an infinite while loop when an exception is encountered. The only way to break out of the loop is to not encounter an exception (which will happen once a connection is established). 

If there is a way to write the test such that the side effect changes after the loop runs once, that's probably the way to go, but I don't know if that is possible.

Reply:
I know you can do that using regular mock but the async nature of this test might make it a little bit more complicated. (I suffered writing it as is 😂) 
I think the way that i did it was each time it was 
```
m.side_effect = ['foo', 'bar', 'baz']
```
So each time that the thing you mock gets called it will return the next item in the list. I know that it might not actually be as straightforward as that since there is async complexity. If possibly - I would try to figure it out bc I do think we are losing value by removing it.

Reply:
Glad you pushed to include the test. I couldn't get the test to finish because of some silly threading I was doing, but I was able to remove the threading all together. Now the connection_check is a lot simpler!

(though I need to make sure all the changes I made still work)

Reply:
Yay! Thanks for fixing that! 😃